ykbh/src/lib.rs

```diff
@@ -142,21 +142,10 @@ impl SIRInterpreter {
 
     /// Given a vector of local declarations, create a new StackFrame, which allocates just enough
     /// space to hold all of them.
-    fn allocate_locals(local_decls: &Vec<LocalDecl>) -> StackFrame {
-        // FIXME Soon this will be pre-computed and handed to us by SIR.
-        let mut offsets = Vec::new();
-        let mut layout = Layout::from_size_align(0, 1).unwrap();
-        for d in local_decls {
-            let align = SIR.ty(&d.ty).align();
-            let size = SIR.ty(&d.ty).size();
-            let l = Layout::from_size_align(size.try_into().unwrap(), align.try_into().unwrap())
-                .unwrap();
-            let (nl, s) = layout.extend(l).unwrap();
-            offsets.push(s);
-            layout = nl;
-        }
-        layout = layout.pad_to_align();
-
+    fn create_frame(body: Arc<Body>) -> StackFrame {
```

Comment:
This is optional, but I think you could make this more general bu getting rid of the `Arc` by making this accept `&body` and then doing `&*` at the call-site?

Reply:
Wouldn't that move the `Body`? I might have had a reason why I did it this way in the last PR, though I can't remember.

Reply:
I'm not sure actually. Might be worth a shot?

Reply:
Better? 4b3c667

Reply:
Great!